app/[locale]/feed/[feed]/route.ts

```diff
@@ -29,7 +29,9 @@ export const GET = async (_: Request, { params }: StaticParams) => {
 export const generateStaticParams = async () =>
   siteConfig.rssFeeds.map(feed => ({ feed: feed.file, locale }));
 
-// Forces that only the paths from `generateStaticParams` are allowed, giving 404 on the contrary
+// In this case we want to catch-all possible requests. This is so that if a non defined feed is
+// requested we can manually return a 404 response for it instead of having Next.js handle it
+// and return our top level custom 404 html page instead
 // @see https://nextjs.org/docs/app/api-reference/file-conventions/route-segment-config#dynamicparams
 export const dynamicParams = true;
```

Comment:
On a side note, I'm quite sure dynamicParams should not be active for blog feeds, but I don't recall now why this is set to true or if it was an accident

Reply:
as I mentioned in the comment, with it set to `true` you don't get the custom 404 html page, and instead you get a 404 with an empty body: 
![Screenshot 2024-03-12 at 22 36 07](https://github.com/nodejs/nodejs.org/assets/61631103/a9e15663-5146-4f88-a96f-586b5a210b97)

my guess is that you didn't want to return some html as this could cause issue with RSS apps? ¯\\_(ツ)_/¯

Reply:
Oh true, I recall now. Indeed because in some endpoints we are self consuming the data, if the 404 returned a HTML page it would break the JSON.parse method 👀

Reply:
I sorta question the decision that it renders a HTML page when dynamicParams is disabled. I understand that for pages that makes sense... But for API routes 🤔

Reply:
yeah.... weird that Pages routing seems to catch also the api requests 😅 🤷